File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/SubtextNode.java
package xyz.wingio.plugins.morehighlight;

import com.discord.simpleast.core.node.Node;
import com.discord.utilities.textprocessing.*;
import com.discord.utilities.color.ColorCompat;

import android.text.SpannableStringBuilder;
import android.text.style.*;
import android.graphics.Color;

import com.lytefast.flexinput.R;

public class SubtextNode<MessageRenderContext> extends Node<MessageRenderContext> {
  String content;

  public SubtextNode(String content){
    super();
    this.content = content;
  }

  @Override
  public void render(SpannableStringBuilder builder, MessageRenderContext renderContext) {
    int length = builder.length();
    builder.append(content);
    
    builder.setSpan(new RelativeSizeSpan(0.85f), length, builder.length(), SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE);
    
    int greyColor = Color.parseColor("#99AAB5");
    builder.setSpan(new ForegroundColorSpan(greyColor), length, builder.length(), SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE);
  }
}

File: MoreHighlight/src/main/java/xyz/wingio/plugins/morehighlight/node/SubtextNode.kt
package xyz.wingio.plugins.morehighlight.node

import android.content.Context
import android.text.SpannableStringBuilder
import android.text.style.ForegroundColorSpan
import android.text.style.RelativeSizeSpan
import com.discord.simpleast.core.node.Node
import com.discord.utilities.color.ColorCompat
import com.lytefast.flexinput.R

class SubtextNode<MessageRenderContext>(
    private val context: Context
): Node.a<MessageRenderContext>() {

    override fun render(builder: SpannableStringBuilder, renderContext: MessageRenderContext) {
        val length = builder.length
        super.render(builder, renderContext)

        val mutedColor = ColorCompat.getThemedColor(context, R.b.colorTextMuted)
        builder.setSpan(RelativeSizeSpan(0.85f), length, builder.length, SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE)
        builder.setSpan(ForegroundColorSpan(mutedColor), length, builder.length, SpannableStringBuilder.SPAN_EXCLUSIVE_EXCLUSIVE)
    }

}
